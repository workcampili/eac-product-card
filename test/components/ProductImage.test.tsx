import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';


describe('ProductImage', () => {
    test('should show the component with  no Image', () => {
        const wrapper = renderer.create(
            <ProductImage img="http://hola.jpg" className="custom-class" />);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should show the component with product title', () => {

        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }


            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });



});