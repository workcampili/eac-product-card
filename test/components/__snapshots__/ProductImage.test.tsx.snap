// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should show the component with  no Image 1`] = `
<img
  alt="Product "
  className="productImg custom-class"
  src="http://hola.jpg"
/>
`;

exports[`ProductImage should show the component with product title 1`] = `
<div
  className="productCard  undefined"
>
  <img
    alt="Product "
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
